src-self-hosted/ir.zig

```diff
@@ -243,16 +243,10 @@ pub const Inst = struct {
             return 2;
         }
         pub fn getOperand(self: *const BinOp, index: usize) ?*Inst {
-            var i = index;
-
-            if (i < 1)
+            if (index == 0)
                 return self.lhs;
-            i -= 1;
-
-            if (i < 1)
+            if (index == 1)
```

Comment:
If we ignored the pattern set by the rest of the file, this should be a switch expression. But I don't think we should ignore the pattern set by the rest of the file